dspace-api/src/test/java/org/dspace/discovery/DiscoveryIT.java

```diff
@@ -0,0 +1,191 @@
+
+        discoverResult = searchService.search(context, discoverQuery);
+        indexableObjects = discoverResult.getIndexableObjects();
+        assertEquals(0, indexableObjects.size());
+        assertEquals(0, discoverResult.getTotalSearchResults());
+    }
+
+    @Ignore
+    @Test
+    public void deleteWorkspaceItemSolrRecordAfterDeletionFromDbTestn() throws Exception {
+        context.turnOffAuthorisationSystem();
+        Community community = CommunityBuilder.createCommunity(context)
+                                              .withName("Parent Community")
+                                              .build();
+        Collection collection = CollectionBuilder.createCollection(context, community)
+                                                 .withWorkflowGroup(1, admin)
+                                                 .build();
+
+        WorkspaceItem wsi = WorkspaceItemBuilder.createWorkspaceItem(context, collection)
+                                                .withTitle("Test item")
+                                                .withIssueDate("2019-03-06")
+                                                .withSubject("ExtraEntry")
+                                                .build();
+
+        ItemBuilder.createItem(context, collection).build();
+
+
+
+        Workflow workflow = XmlWorkflowServiceFactory.getInstance().getWorkflowFactory().getWorkflow(collection);
+
+        ItemBuilder.createItem(context, collection).build();
+
+        context.restoreAuthSystemState();
+
+        MockHttpServletRequest httpServletRequest = new MockHttpServletRequest();
+        httpServletRequest.setParameter("submit_approve", "submit_approve");
+
+        XmlWorkflowItem workflowItem = workflowService.startWithoutNotify(context, wsi);
+        context.dispatchEvents();
+        indexer.commit();
+
+        assertSearchQuery(IndexablePoolTask.TYPE, 1);
+
+        context.turnOffAuthorisationSystem();
+        ItemBuilder.createItem(context, collection).build();
+        context.restoreAuthSystemState();
+
+        executeWorkflowAction(httpServletRequest, admin, workflow, workflowItem,
+                "reviewstep", "claimaction");
+
+
+        context.dispatchEvents();
+        indexer.commit();
+
+        context.turnOffAuthorisationSystem();
+        ItemBuilder.createItem(context, collection).build();
+        context.restoreAuthSystemState();
+
+        assertSearchQuery(IndexablePoolTask.TYPE, 0);
+        assertSearchQuery(IndexableClaimedTask.TYPE, 1);
+
+        returnToPool(admin, workflowItem);
+        context.dispatchEvents();
+        indexer.commit();
+
+        context.turnOffAuthorisationSystem();
+        ItemBuilder.createItem(context, collection).build();
+        context.restoreAuthSystemState();
+
+        assertSearchQuery(IndexablePoolTask.TYPE, 1);
+        assertSearchQuery(IndexableClaimedTask.TYPE, 0);
+
+        workflowService.deleteWorkflowByWorkflowItem(context, workflowItem, admin);
```

Comment:
It doesn't appear that we are testing that everything is (in fact) deleted.  It looks like we are just testing that the deletion succeeds.  Am I misunderstanding the purpose of this test?  If so, perhaps we can add more inline comments here to describe what this test is proving.

It's also unclear (to me) why we keep creating new Items in this test via the ItemBuilder (see lines 152 and 141).... so, it may be good to add inline comments here in general, as it's unclear why adding additional Items is necessary to test the number of Pooled or Claimed Tasks in the Workflow.